src/openlcb/CanFilter.hxx

```diff
@@ -0,0 +1,310 @@
+/** \copyright
+ * Copyright (c) 2024, Balazs Racz
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file CanFilter.hxx
+ *
+ * Filter class for routing CAN frames.
+ *
+ * @author Balazs Racz
+ * @date 27 Jan 2024
+ */
+
+#ifndef _OPENLCB_CANFILTER_HXX_
+#define _OPENLCB_CANFILTER_HXX_
+
+#include <algorithm>
+#include <cstdint>
+#include <map>
+#include <vector>
+
+#include "os/OS.hxx"
+#include "openlcb/CanDefs.hxx"
+#include "utils/Hub.hxx"
+
+namespace openlcb
+{
+
+/**
```

Comment:
A little unexpected to see 'C' style comment blocks instead of C++ style for new code.

Reply:
ack.
I don't have a tool to reformat them automatically so i'll leave them as is.